apps/dashboard/app/javascript/packs/files/UppyOps.js

```diff
@@ -0,0 +1,172 @@
+import { Uppy, BasePlugin } from '@uppy/core'
+import Dashboard from '@uppy/dashboard'
+import XHRUpload from '@uppy/xhr-upload'
+import _ from 'lodash';
+
+
+let uppy = null;
+
+$(document).ready(function() {

```

Comment:
It can be declared out of document ready, but should it?  My thoughts are that we should make sure the document loads and is ready prior to initializing unless we have to.  uppy was a 1 off due to the need to be in a global scope for the file.

Is there a good reason why it should not be in document.ready?

Reply:
> It can be declared out of document ready, but should it?

Which way is more advantageous? I'd say having one `const uppy` is better than initializing the variable to null then modifying it after the fact, just because it's simpler. Less state to worry about.

Reply:
If you are just talking about the uppy file, then that's fine. I didn't really need to do any major updates to that file.  So, to be consistent I threw most of the stuff into document.ready.

Reply:
I was talking about this specifically, but it does seem like an odd pattern.

I know that we need to do this sometime because files are accessing variables declared somewhere else, but when we don't _need_ to do it, I'd argue that simpler is better.

Reply:
I move the stuff out of document.ready and things broke. I now remember why I put it there.  There are variables that have must live in the template file since the values are coming from configs and ruby is reading those values.